qxf2/qxf2-lambdas
# Issue 94 url filter fix

- Added fix for issue 94 "Method post_to_newsletter returns a variable that could never be declared"
- Files changed:
- tests/test_url_filter.py
- tests/test_url_filter.py

What has been changed:
-  Added a check for empty lists in post_newsletter() method as well, and returning the values appropriately.
- Changing the test_status_code appropriately based on if the list is empty or not

How to test the fix:
1. First switch to branch `issue-94-url-filter`
2. Run `pytest tests/test_url_filter.py -sv`
3. You'll see one of the unit test fails with `UnboundLocalError: local variable 'response' referenced before assignment` error
4. Now switch to `issue-94-url-filter-fix` branch
5. Now run `pytest tests/test_url_filter.py -sv`
6. You should notice that all the tests pass including the previous test which had failed
